src/status_im/contexts/wallet/events.cljs

```diff
@@ -547,3 +551,36 @@
  :wallet/process-watch-only-account-from-backup
  (fn [_ [{:keys [backedUpWatchOnlyAccount]}]]
    {:fx [[:dispatch [:wallet/process-account-from-signal backedUpWatchOnlyAccount]]]}))
+
+(defn process-keypairs
+  [{:keys [db]} [keypairs]]
+  (let [existing-keypairs-by-id               (get-in db [:wallet :keypairs])
+        existing-accounts-by-address          (get-in db [:wallet :accounts])
+        {:keys [removed-keypair-ids
+                removed-account-addresses
+                updated-keypairs-by-id
+                updated-accounts-by-address]} (data-store/process-keypairs keypairs)
+        updated-keypair-ids                   (set (keys updated-keypairs-by-id))
+        updated-account-addresses             (set (keys updated-accounts-by-address))
+        existing-account-addresses            (set (keys existing-accounts-by-address))
+        new-account-addresses                 (clojure.set/difference updated-account-addresses
+                                                                      existing-account-addresses)
+        old-account-addresses                 (->> (vals existing-accounts-by-address)
+                                                   (filter (fn [{:keys [address key-uid]}]
+                                                             (and (contains? updated-keypair-ids key-uid)
+                                                                  (nil? (get updated-accounts-by-address
+                                                                             address)))))
+                                                   (map :address))]
+    (cond-> {:db (-> db
+                     (assoc-in [:wallet :keypairs]
+                               (-> (partial dissoc existing-keypairs-by-id)
```

Comment:
I had to stop and think for a while to understand `partial` with dissoc and the next `apply` threaded. Could be simply:

```clojure
(-> db
    (assoc-in [:wallet :keypairs]
              (-> (apply dissoc existing-keypairs-by-id removed-keypair-ids)
                  (merge updated-keypairs-by-id))))
```

Reply:
I think apply expects a `seq` as the second argument (?), so I don't think I can write it like that, but I'll try again to see if it works, happy to clean this up 🧹

Reply:
Ah okay, I thought `removed-keypair-ids` was a seq/nil, then it would work.

Reply:
`removed-keypair-ids` is a seq, but I think I misunderstood how `apply` works. I just tested your suggestion and the tests pass, so I'll update the code to remove the `partial` expression.